### ISSUE TITLE
wasm-transformer: output is invalid

### ISSUE THREAD

**Issue by corwin-of-amber** [wasmerio/wasmer-js#152]
Running the attached WASM binary through lowerI64Imports resulted in an invalid file. Wasmtime, wasm2wat, wasmer, wasmer-js all report that it is corrupted. Trying to `WebAssembly.instantiate()` the result gives `unknown section code #0x75 @+530`.

I was using the most recent version from master at the time of this post. This is the code snippet based on @wasmer/cli.

```js
const { lowerI64Imports } = require('@wasmer/wasm-transformer');

const fs = require('fs');

var bytes = fs.readFileSync('busy.wasm');
bytes = lowerI64Imports(bytes);

fs.writeFileSync('busy32.wasm', bytes);
```

[busy.zip](https://github.com/wasmerio/wasmer-js/files/3798848/busy.zip)

**Comment by corwin-of-amber** [comment]
I forgot to mention that `busy.wasm` was compiled using wasi-sdk-7.

**Comment by torch2424** [comment]
Ah, yeah so they are some, WASI modules that still give us trouble that we've been working through. The transformations are done in a way that makes debugging a bit difficult since we don't create an actual AST. We will try and debug this once we get the chance 😄 

Thank you very much for opening the issue, it helps us to get as many modules as we can. Also, is the source code for the Wasm module the same from the email I got a while back? 😄 

cc @syrusakbary

**Comment by corwin-of-amber** [comment]
Hi, no, that one did not call any APIs with i64 in their type. This program is calling `opendir`.
It works with `fopen`, just not `opendir` 😕 

[busy.c.txt](https://github.com/wasmerio/wasmer-js/files/3799558/busy.c.txt)

**Comment by corwin-of-amber** [comment]
I was able to circumvent the error for now using this plug:
```c
#include <wasi/core.h>


#define __WASI_SYSCALL_NAME(name) \
    __attribute__((__import_module__("wasi_unstable"), __import_name__(#name)))


__wasi_errno_t __wasi_fd_readdir$32(
    __wasi_fd_t fd,
    void *buf,
    size_t buf_len,
    int32_t cookie,  /* <-- wrapped to 32 */
    size_t *bufused
) __WASI_SYSCALL_NAME(fd_readdir) __attribute__((__warn_unused_result__));


__wasi_errno_t __wasi_fd_readdir(
    __wasi_fd_t fd,
    void *buf,
    size_t buf_len,
    __wasi_dircookie_t cookie,
    size_t *bufused
) {
    return __wasi_fd_readdir$32(fd, buf, buf_len, cookie, bufused);
}


#undef __WASI_SYSCALL_NAME
```

That's not a solution of course, but it least it shows that the problem lies with the `$__wasi_fd_readdir` import.

**Comment by torch2424** [comment]
@corwin-of-amber 

> Hi, no, that one did not call any APIs with i64 in their type. This program is calling opendir. The transformer works when calling fopen, just not opendir 😕

Huh? That is quite odd. I'll take a look once I get the chance 😄 Also, thanks for providing the source code example!

> I was able to circumvent the error for now using this plug:

Oh that's even more odd, I wonder what that import is doing? This helps us isolate a test case by a ton thoough! Thank you! 😄

**Comment by syrusakbary** [comment]
Investigating this now

**Comment by torch2424** [comment]
This is now fixed in `0.4.7`.

Thank you @syrusakbary for finding this bug! 😄 🎉 

Thank you @corwin-of-amber for filing the issue, and being patient for the fix! 😄 👍
